ADJ-Internship/BusinessService/Implementations/ProgressCheckService.cs

```diff
@@ -0,0 +1,204 @@
+﻿using ADJ.BusinessService.Core;
+using ADJ.BusinessService.Dtos;
+using ADJ.BusinessService.Interfaces;
+using ADJ.Common;
+using ADJ.DataModel.OrderTrack;
+using ADJ.Repository.Core;
+using ADJ.Repository.Interfaces;
+using AutoMapper;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace ADJ.BusinessService.Implementations
+{
+    public class ProgressCheckService : ServiceBase, IProgressCheckService
+    {
+        private readonly IDataProvider<Order> _odDataProvider;
+        private readonly IOrderRepository _odRepository;
+
+        private readonly IOrderDetailRepository _oddRepository;
+        private readonly IDataProvider<OrderDetail> _oddDataProvider;
+
+        private readonly IDataProvider<ProgressCheck> _prcDataProvider;
+        private readonly IProgressCheckRepository _prcRepository;
+
+        public ProgressCheckService(IUnitOfWork unitOfWork, IMapper mapper, ApplicationContext appContext,
+            IDataProvider<ProgressCheck> prcDataProvider,IDataProvider<Order> odDataProvider,
+            IDataProvider<OrderDetail> oddDataProvider, IProgressCheckRepository prcRepository,
+            IOrderRepository odRepository,IOrderDetailRepository oddRepository) : base(unitOfWork, mapper, appContext)
+        {
+            _odDataProvider = odDataProvider;
+            _odRepository = odRepository;
+
+            _oddRepository = oddRepository;
+            _oddDataProvider = oddDataProvider;
+
+            _prcRepository = prcRepository;
+            _prcDataProvider = prcDataProvider;
+        }
+        public async Task<int> CheckOrderHaventProgress(int orderId)
+        {
+            var result = await _prcDataProvider.ListAsync();
+            List<ProgressCheck> prcList = result.Items;
+            int check = 0;
+            foreach (var item in prcList)
+            {
+                if (item.OrderId == orderId)
+                {
+                    check += 1;                  
+                }
+            }
+            return check;
+        }
+        public void CreateDefaultModel(int orderId)
+        {
+            ProgressCheck progressCheck = new ProgressCheck
```

Comment:
what is this method for?

Reply:
when order hasn't progress check i will create for this one default progress check object
.